### DISCUSSION TITLE
Wheel motions for a mobile manipulator model

### DISCUSSION THREAD

**Discussion by LHT-Liu** [stack-of-tasks/pinocchio#2206] [Q&A]
hello,my model is mobile manipulator,as shown in the following picture.
![image](https://github.com/stack-of-tasks/pinocchio/assets/64388319/5b0dc166-58ca-43a6-bf42-6587c2d9048b)
I know that there is a [planar](https://gepettoweb.laas.fr/doc/stack-of-tasks/pinocchio/master/doxygen-html/md_doc_a-features_intro.html#autotoc_md16) joint type in pinocchio, but during simulation, I would like the wheels to move realistically. Should I build four wheels as rotating joints, or treat them as a whole and use plannar joints directly?

thanks

**Comment by stephane-caron** [comment]
You can set a planar joint for the mobile base and revolute joints at the wheels, then apply the [wheel task](https://placo.readthedocs.io/en/latest/kinematics/wheel_task.html) from PlaCo. There is a full example in [`omniwheel.py`](https://github.com/Rhoban/placo-examples/blob/0dd743a4a61d066678388c9031fe4537dd31bfbb/kinematics/omniwheel.py) that produces this behavior:

https://github.com/stack-of-tasks/pinocchio/assets/1189580/567f16cf-7b70-4e2c-aa83-db561e238961

PlaCo builds upon Pinocchio but it is a separate project. If you have questions about it, you can ask them in the [PlaCo issue tracker](https://github.com/Rhoban/placo/issues).
